# HG changeset patch
# User Rik <rik@octave.org>
# Date 1449094149 28800
#      Wed Dec 02 14:09:09 2015 -0800
# Node ID 62564952e161fa58a1a230d47463398e3190d0f3
# Parent  b6f2909e7f94e8d60260c2e64385c662668c7e25
# Parent  a8ee668e7fd7cae9b821bc9b8676df46ff29db13
maint: Periodic merge of stable to default.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2271,17 +2271,17 @@ AC_CHECK_FUNCS([getegid geteuid getgid g
 AC_CHECK_FUNCS([getpgrp getpid getppid getpwent getpwuid getuid])
 AC_CHECK_FUNCS([isascii kill])
 AC_CHECK_FUNCS([lgamma lgammaf lgamma_r lgammaf_r])
 AC_CHECK_FUNCS([log1p log1pf pipe])
 AC_CHECK_FUNCS([realpath resolvepath roundl])
 AC_CHECK_FUNCS([select setgrent setpwent siglongjmp strsignal])
 AC_CHECK_FUNCS([tcgetattr tcsetattr tgammaf toascii])
 AC_CHECK_FUNCS([umask waitpid])
-AC_CHECK_FUNCS([_kbhit])
+AC_CHECK_FUNCS([_getch _kbhit])
 
 dnl There are no workarounds in the code for missing these functions.
 AC_CHECK_FUNCS([modf pow sqrt sqrtf], [],
                [AC_MSG_ERROR([Missing function required to build Octave])])
 
 ## exp2, round, tgamma function checks
 AC_LANG_PUSH(C++)
 AC_CHECK_DECLS([exp2, round, tgamma], [], [], [[#include <cmath>]])
diff --git a/libinterp/corefcn/sysdep.cc b/libinterp/corefcn/sysdep.cc
--- a/libinterp/corefcn/sysdep.cc
+++ b/libinterp/corefcn/sysdep.cc
@@ -503,18 +503,25 @@ octave_pclose (FILE *f)
 #endif
 }
 
 // Read one character from the terminal.
 
 int
 octave_kbhit (bool wait)
 {
-#ifdef HAVE__KBHIT
-  int c = (! wait && ! _kbhit ()) ? 0 : std::cin.get ();
+#ifdef HAVE__KBHIT && HAVE__GETCH
+  // This essentially means we are on a Windows system.
+  int c;
+
+  if (wait)
+    c = _getch ();
+  else
+    c = (! _kbhit ()) ? 0 : _getch ();
+
 #else
   raw_mode (true, wait);
 
   // Get current handler.
   octave_interrupt_handler saved_interrupt_handler
     = octave_ignore_interrupts ();
 
   // Restore it, disabling system call restarts (if possible) so the
